test/scripts/run-gateway-conformance.sh

```diff
@@ -41,8 +41,22 @@ spec:
   controllerName: projectcontour.io/gateway-controller
 EOF
 
-cd $(mktemp -d)
-git clone https://github.com/kubernetes-sigs/gateway-api
-cd gateway-api
-git checkout "${GATEWAY_API_VERSION}"
-go test ./conformance -gateway-class=contour
+
+# If we're running conformance tests for the same Gateway API version
+# that we're using via go.mod, use our own test driver (via `go test`)
+# so we can opt into additional supported features that we know we support.
+# Otherwise, we're likely running the `main` conformance tests for a nightly
+# build, where we have to clone the upstream repo to be able to run that
+# version of the tests, but lose the ability to opt into tests for additional
+# supported features since that's not exposed via flag.
+GO_MOD_GATEWAY_API_VERSION=$(grep "sigs.k8s.io/gateway-api" go.mod | awk '{print $2}')
+
+if [ "$GATEWAY_API_VERSION" = "$GO_MOD_GATEWAY_API_VERSION" ]; then
+  go test -tags conformance ./test/conformance/gatewayapi --gateway-class=contour
+else 
+  cd $(mktemp -d)
+  git clone https://github.com/kubernetes-sigs/gateway-api
+  cd gateway-api
+  git checkout "${GATEWAY_API_VERSION}"
+  go test ./conformance -gateway-class=contour
+fi
```

Comment:
Re: https://github.com/projectcontour/contour/issues/4616#issuecomment-1185932732, this seems like the best option for now, as I definitely want to (a) opt into the tests for extended features that we support; and (b) continue running new `main` conformance tests on a nightly basis as they're added upstream.

Reply:
do we have somewhere we explicitly set `GATEWAY_API_VERSION`, from https://github.com/projectcontour/contour/blob/e895c86a85eb6ba076276fe04dea83ddc029bc29/Makefile#L10 seems like it gets set by the version in the module too so they won't diverge? I'm probably missing some place where this works when we bump the version on `main` or something

Reply:
Yeah, for the daily build that runs the conformance tests from `main` it's set here:
https://github.com/projectcontour/contour/blob/main/.github/workflows/build_daily.yaml#L141-L144

Reply:
oh duh, was looking at the wrong workflow file 🙃